geoh5py/workspace/workspace.py

```diff
@@ -103,19 +103,29 @@ def __init__(self, h5file: str = "Analyst.geoh5", **kwargs):
             for key, attr in proj_attributes.items():
                 setattr(self, self._attribute_map[key], attr)
 
-            # Get the Root attributes
-            attributes, type_attributes, _ = H5Reader.fetch_attributes(
-                self.h5file, uuid.uuid4(), "Root"
-            )
-            self._root = self.create_entity(
-                RootGroup, save_on_creation=False, **{**attributes, **type_attributes}
-            )
+            try:
+                self._root = self.load_entity(uuid.uuid4(), "root")
```

Comment:
**suggestion:**  have a dedicated function (`fetch_or_create_root()` ?) that encapsulates the whole logic about getting or creating root

Reply:
Done